### PR TITLE
[Fix  #1779] New cop ModuleLength

### LINKED ISSUE #1779: Why isn't there a Metrics/ModuleLength cop?
There are both ClassLength and MethodLength but no ModuleLength.

What is the rational behind this? Am I the only one to stumble upon modules too fat for their own good?

### PR DESCRIPTION
**Pull request by sdeframond**
Following discussion on #1779, here is a new cop that checks for overly long module definitions.

Since `ModuleLength` and `ClassLength` do almost the same thing, I grouped the common code into a new `ClassishLength` module.

Hope this helps.

### CODE REVIEW

**Comment by sdeframond** [comment]
Oops, seems like it has some issues. Please do not merge yet.

**Comment by sdeframond** [comment]
Looking better now :smile:. Go ahead!

**Comment by jonas054** [comment]
Please change the commit message to "[Fix #1779] New cop ModuleLength" (no dot at the end).

**Comment by sdeframond** [comment]
Should I change the changelog entry as well? 

On April 12, 2015 6:34:13 PM CEST, Jonas Arvidsson <notifications@github.com> wrote:
>Please change the commit message to "[Fix #1779] New cop ModuleLength"
>(no dot at the end).
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/bbatsov/rubocop/pull/1788#issuecomment-92085680

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sdeframond** [comment]
Done.

**Comment by jonas054** [comment]
:+1: Looks good to me!

**Comment by sdeframond** [comment]
Hi, is there something blocking this feature? Should I rebase it on master?

**Comment by jonas054** [comment]
@sdeframond You need to rebase now.

**Comment by sdeframond** [comment]
@jonas054 I just rebased.

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 ### New features
 
+* [#1788](https://github.com/bbatsov/rubocop/pull/1788): New cop `ModuleLength` checks for overly long module definitions. ([@sdeframond][])
```
Comment (bbatsov) [inline CHANGELOG.md]: This entry is not in the right place.

Comment (sdeframond) [inline CHANGELOG.md]: Oops, my bad. It should be OK now.

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**